File: csharp/1.cs
using System;

namespace Sample
{
	internal class Program
	{
		private static void Main(string[] args)
		{
			Console.Write("Masukkan jumlah permen yang dibeli: ");
			var permen = int.Parse(Console.ReadLine());

			var bonus = Math.Floor(permen / 5.0);
			var total_bonus = bonus;
			while (bonus > 0)
			{
				bonus = Math.Floor(bonus / 5.0);
				total_bonus += bonus;

				if (bonus < 4)
				{
					break;
				}
			}

			Console.WriteLine($"Bonus permen yang didapat: { total_bonus }");
		}
	}
}

File: cpp/1.cpp
#include <stdio.h>

int main()
{
    int n;
	printf("n: ");
	scanf("%d", &n);

    int bonus = n / 5.0;

    int temp_bonus = bonus;
    int bonus_tambahan = 0;
    while (temp_bonus > 4)
    {
        bonus_tambahan = bonus_tambahan + temp_bonus / 5.0;
        temp_bonus = temp_bonus / 5.0;
    }

    printf("Bonus permen yang didapat: %d", bonus + bonus_tambahan);

    return 0;
}
